app/lib/core/models/acknowledgement_info.dart

```diff
@@ -12,4 +12,11 @@ class AcknowledgementInfo {
       {@required this.shipperAck,
       @required this.transporterAck,
       @required this.receiverAck});
+
+  // TODO: Resolve how to display ack info
+  // Most other info is strings, perhaps this should give a widget for display
+  String toString() => '''Shipper acknowledgement:
```

Comment:
The more I look at these methods the more I hate them. We will absolutely be extracting these all out to their respective display widget (which increases the amount of code to maintain, but the display and object are 1-to-1 so changing one forces changes on the other) (I'm still not sure of a better way to do this...).

[Expandable list of widgets example.](https://stackoverflow.com/questions/50530152/how-to-create-expandable-listview-in-flutter)

Reply:
@MummenRider should I do this right away, or move on?